### PR TITLE
Fix parallel codegen and parallel scan, force insert/update/delete to be serial.

### PR DESCRIPTION
**Pull request by lmwnshn**
Previously, we accidentally set codegen to always generate serial queries.

Fixing codegen exposed some bugs in parallel scan translation, which have been fixed.
Inserts, updates, and deletes are serial for now.

### CODE REVIEW

**Comment by lmwnshn** [comment]
For CLion users, File -> Settings -> Dynamic Analysis Tools -> Sanitizers -> Leak Sanitizer, add

suppressions=" ... /CLionProjects/terrier/build-support/data/lsan_suppressions.txt"

File: test/execution/sql_table_vector_iterator_test.cpp [hunk 1]
```diff
@@ -135,13 +135,16 @@ TEST_F(TableVectorIteratorTest, ParallelScanTest) {
 
   struct Counter {
     uint32_t c_;
+    exec::ExecutionContext *exec_ctx_;
```
Comment (pmenon) [inline test/execution/sql_table_vector_iterator_test.cpp:138]: Do you need this?

File: src/include/execution/exec/execution_settings.h [hunk 2]
```diff
@@ -27,7 +27,7 @@ class EXPORT ExecutionSettings {
   }
 
   /** @return True if parallel query execution is enabled. */
-  constexpr bool GetIsParallelQueryExecution() const { return is_parallel_query_execution_; }
+  constexpr bool GetIsParallelQueryExecutionEnabled() const { return is_parallel_query_execution_; }
```
Comment (pmenon) [inline src/include/execution/exec/execution_settings.h:30]: Nit: rename to `is_parallel_execution_enabled_` ? As written, it's a little unclear, though function name clears it up.

File: test/execution/sql_table_vector_iterator_test.cpp (hunk 1 above)
Comment (lmwnshn) [reply test/execution/sql_table_vector_iterator_test.cpp]: Removed

File: src/include/execution/exec/execution_settings.h (hunk 2 above)
Comment (lmwnshn) [reply src/include/execution/exec/execution_settings.h]: Renamed

**Pull request merged**